java-frontend/src/main/java/org/sonar/java/se/symbolicvalues/BinarySymbolicValue.java

```diff
@@ -37,10 +48,24 @@ public boolean references(SymbolicValue other) {
 
   @Override
   public void computedFrom(List<SymbolicValue> symbolicValues) {
-    Preconditions.checkArgument(symbolicValues.size() == 2);
+    List<ProgramState.SymbolicValueSymbol> symbolicValueSymbols = symbolicValues.stream()
+      .map(sv -> new ProgramState.SymbolicValueSymbol(sv, null))
+      .collect(Collectors.toList());
+    computedFromSymbols(symbolicValueSymbols);
+  }
+
+  @Override
+  public void computedFromSymbols(List<ProgramState.SymbolicValueSymbol> valueSymbols) {
+    Preconditions.checkArgument(valueSymbols.size() == 2);
     Preconditions.checkState(leftOp == null && rightOp == null, "Operands already set!");
-    rightOp = symbolicValues.get(0);
-    leftOp = symbolicValues.get(1);
+    rightOp = valueSymbols.get(0).symbolicValue();
+    rightSymbol = valueSymbols.get(0).symbol();
+    leftOp = valueSymbols.get(1).symbolicValue();
+    leftSymbol = valueSymbols.get(1).symbol();
+  }
+
+  public List<Symbol> computedFromSymbols() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the "@Override" annotation above this method signature [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1161)